### PR TITLE
Add Android Jetpack Compose example

### PR DESCRIPTION
**Pull request by Bencodes**
Adding an Android Jetpack Compose example that showcases how to set up Jetpack Compose.

### CODE REVIEW

**Comment by Bencodes** [comment]
This is going to be blocked by this for now https://github.com/bazelbuild/continuous-integration/issues/1494

**Review by ahumesky — APPROVED** [review]
Looks good!

File: android/jetpack-compose/WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    sha256 = "cd06d15dd8bb59926e4d65f9003bfc20f9da4b2519985c27e190cddc8b7a7806",
+    strip_prefix = "rules_android-0.1.1",
+    urls = ["https://github.com/bazelbuild/rules_android/archive/v0.1.1.zip"],
+)
+
+load("@build_bazel_rules_android//android:rules.bzl", "android_sdk_repository")
+
+android_sdk_repository(
+    name = "androidsdk",
+    api_level = 31,
```
[the remark this answers is not in the document]
Comment (Bencodes) [reply android/jetpack-compose/WORKSPACE:66]: It is needed, and we should revert the explicit api_level before we merge as well.

I filed an issue over [here](https://github.com/bazelbuild/continuous-integration/issues/1494) to have 31 installed on the mac hosts, which didn't seem to happen back when we first added 31 support.

[the remark this answers is not in the document]
Comment (Bencodes) [reply android/jetpack-compose/WORKSPACE:66]: The other host types already have 31 installed, so worst case we can disable the mac os jobs and enable them when we are unblocked.

[the remark this answers is not in the document]
Comment (ted-xie) [reply android/jetpack-compose/WORKSPACE:66]: The issue with the CI machines has been fixed now: https://github.com/bazelbuild/continuous-integration/issues/1494#issuecomment-1315876482

**Pull request merged**